Scuti/Scripts/UI/Store/Offer Details/UIPanningAndPinchImage.cs

```diff
@@ -0,0 +1,334 @@
+                Vector2 touchZeroPrevPos = touchZero.position - touchZero.deltaPosition;
+                Vector2 touchOnePrevPos = touchOne.position - touchOne.deltaPosition;
+
+                averagePointBetweenTouch = (touchZeroPrevPos + touchOnePrevPos) / 2;
+
+                Debug.DrawLine(Vector3.zero, touchZero.position, Color.red);
+                Debug.DrawLine(Vector3.zero, touchOne.position, Color.blue);
+                Debug.DrawLine(Vector3.zero, averagePointBetweenTouch, Color.yellow);
+
+
+                // Find the magnitude of the vector (the distance) between the touches in each frame.
+                float prevTouchDeltaMag = (touchZeroPrevPos - touchOnePrevPos).magnitude;
+                float touchDeltaMag = (touchZero.position - touchOne.position).magnitude;
+
+                // Find the difference in the distances between each frame.
+                float deltaMagnitudeDiff = prevTouchDeltaMag - touchDeltaMag;
+                
+                MouseDown(cachedEventData);
+
+                // For scale image
+                var delta = Vector3.one * (deltaMagnitudeDiff * -zoomSpeed);
+                var desiredScale = transform.localScale + delta;
+                desiredScale = ClampDesiredScale(desiredScale);
+
+                transform.localScale = desiredScale;
+
+            }
+
+#if UNITY_EDITOR || UNITY_STANDALONE_WIN
+            if (!isTestingMobile)
+            {
+                // For desktop version
+                if (isEnterToImage)
+                {
+                    MouseDown(cachedEventData);
+                }
+            }
+
+#endif
+        }
+
+        // -------------------------------------------------------------------------------------
+
+        /// <summary>
+        /// This method will be called on the start of the mouse drag
+        /// </summary>
+        /// <param name="eventData">mouse pointer event data</param>
+        public void OnBeginDrag(PointerEventData eventData)
+        {
+            Debug.Log("Begin Drag");
+            lastMousePosition = eventData.position;
+            averagePointBetweenTouch = eventData.position;
+        }
+
+        /// <summary>
+        /// This method will be called during the mouse/touch drag
+        /// </summary>
+        /// <param name="eventData">mouse pointer event data</param>
+        public void OnDrag(PointerEventData eventData)
+        {
+            isDragDetected = true;
+            scrollRect.enabled = false;
+
+            Vector2 currentMousePosition = new Vector2(0, 0);
+
+            // Get de position touch
+            if (counterTouch == 1)
+            {
+                currentMousePosition = eventData.position;
+            }
+
+            if (counterTouch >= 2)
+            {
+                currentMousePosition = averagePointBetweenTouch;
+            }
+
+            Vector2 diff = currentMousePosition - lastMousePosition;
+            RectTransform rect = GetComponent<RectTransform>();
```

Comment:
This needs to be a serialized property and is assigned in the editor. Never call GetComponent in something that is going to be called repeated like an update loop or in this case on Drag.

Reply:
Done